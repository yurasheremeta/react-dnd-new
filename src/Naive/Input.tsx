import * as React from 'react';

interface InputProps {
    value: string
    onChange: any,

}
class Input extends React.Component<InputProps>{
    render() {
        return (
            <input type="text" value={this.props.value} onChange={this.props.onChange} />
        )
    }
}
export default Input;