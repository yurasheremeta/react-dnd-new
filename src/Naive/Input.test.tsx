import * as React from 'react';
import { shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import Input from './Input';
import 'jsdom-global/register';

describe('Input tests', () => {

    it('Render correctly input' , () => {
        const wrapper = renderer.create(<Input value={"aaaaa"} onChange={() => {}}/>).toJSON();
        expect(wrapper).toMatchSnapshot();
    })

    it('Render right value' , () => {
        const defaultProps = {
            value: "Hello",
            onChange: () => {}
        }
        const wrapper = mount(<Input {...defaultProps} />);

        expect(wrapper.prop('value')).toEqual("Hello");
    })
})

