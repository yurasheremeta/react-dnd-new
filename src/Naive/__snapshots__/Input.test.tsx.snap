// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input tests Render correctly input 1`] = `
<input
  onChange={[Function]}
  type="text"
  value="aaaaa"
/>
`;
